finalPrep/modZeroHero.js

```diff
@@ -56,29 +104,95 @@ var excuse = 'I think I forgot to lock up my 1992 Toyota Coralla. Be right back.
 //   - luckyNumbers (array)
 //   - address (object with following key/values: number , street , state, zip)
 
+var scaryMonster = {
+  name : "Evilboy",
+  smell : "Evil",
+  weightInLbs : 700,
+  citiesDestroyed : ["Denver", "Dallas", "Houston"],
+  luckyNumbers : [17, 27, 7],
+  address : [17, "Wallaby Way", "Texas", 75002]
+  }
+
+  console.log(scaryMonster);
+
+
 
 // Create a new class called SuperHero
 // - Your class should have the following DYNAMIC values
-//   - name 
+//   - name
 //   - superpower
-//   - age 
+//   - age
 // - Your class should have the following STATIC values
 //   - archNemesis, assigned to "The Syntax Error"
 //   - powerLevel = 100
-//   - energyLevel = 50 
+//   - energyLevel = 50
+
+class SuperHero {
+  constructor(name, superpower, age, archNemesis, powerLevel, energyLevel){
+    this.name = name;
+    this.superpower = superpower;
+    this.age = age;
+    this.archNemesis = "The Syntax Error";
+    this.powerLevel = 100;
+    this.energyLevel = 50;
+  }
+
+  sayName(){
+    console.log(this.name);
+  }
+
+  maximizeEnergy(){
+    console.log(this.energyLevel = 1000);
+  }
+
+  gainPower(addPower){
+    console.log(this.powerLevel = this.powerLevel + addPower);
+  }
+}
+
 
 // - Create the following class methods
 //   - sayName, should print the hero's name to the console
 //   - maximizeEnergy, should update the energyLevel to 1000
 //   - gainPower, should take an argument of a number and INCREASE the powerLevel by that number
 
 // - Create 2 instances of your SuperHero class
+var goodBoy = new SuperHero("GoodBoy", "Slime Hands", 27);
+goodBoy.sayName();
+
+console.log(goodBoy);
+
+goodBoy.maximizeEnergy();
+
+console.log(goodBoy);
+
+goodBoy.gainPower(50);
+
+console.log(goodBoy);
 
 
+var betterBoy = new SuperHero("BetterBoy", "Slime Feet", 28);
+betterBoy.sayName();
+
+console.log(betterBoy);
+
+betterBoy.maximizeEnergy();
+
+console.log(betterBoy);
+
+betterBoy.gainPower(70);
+
+console.log(betterBoy);
+
 // Reflection
 // What parts were most difficult about this exerise?
 
+//The class methods were most difficult for me, but I got them in the end!
+
 // What parts felt most comfortable to you?
 
+// Everything except the class methods.
+
 // What skills do you need to continue to practice before starting Mod 1?
 
+// Everything, but especially class methods!
```

Comment:
And that's ok!! That is by far the most advanced concept covered here, and you will have LOTS of instruction and practice in Mod 1. It is not expected you are fluent with these when you enter Mod 1.